tests/functions_tests/test_arrays.py

```diff
@@ -331,3 +331,30 @@ def __call__(self, *xs):
                   shape in self.in_shapes]
 
         self.expect(model, xs, name=self.name)
+
+
+class TestDynamicReshape(ONNXModelTest):
+
+    def test_output(self):
+        from onnx_chainer.replace_func import as_funcnode
+
+        class Model(chainer.Chain):
+            def __init__(self):
+                super().__init__()
+
+            @as_funcnode('Reshape')
+            def dynamic_reshape(self, x, shape):
+                # shape is expected as variable type
+                return x.array.reshape(shape.array)
```

Comment:
Maybe better to use `F.reshape` or `x.reshape` so this function can be used even in training mode?